packages/plugin-http-client/src/Processor.ts

```diff
@@ -0,0 +1,28 @@
+import { HttpAgentResponse, Dependencies } from '@ima/core';
+
+import { HttpClientRequest } from './HttpClient';
+
+export enum Operation {
+  PRE_REQUEST = 'preRequest',
+  POST_REQUEST = 'postRequest',
+}
+
+export type ProcessorParams<B> = {
+  response: HttpAgentResponse<B> | null;
+  request: HttpClientRequest;
+  additionalParams?: any;
+};
+
+export class Processor {
+  static get $dependencies(): Dependencies {
+    return [];
+  }
+
+  preRequest<B>(params: ProcessorParams<B>) {
```

Comment:
I am not really sure how the `B` passing will work in the real world usage... I would probably use default `B = any` for situations, where you don't have enough typed stuff and don't want to specify the body.

Reply:
I hope I changed it on all places.